packages/docregistry/src/context.ts

```diff
@@ -660,9 +660,11 @@ export class Context<
    */
   private _revert(initializeModel: boolean = false): Promise<void> {
     const opts: Contents.IFetchOptions = {
-      format: this._factory.fileFormat,
       type: this._factory.contentType,
-      content: this._factory.fileFormat !== null
+      content: this._factory.fileFormat !== null,
+      ...(this._factory.fileFormat !== null
+        ? { format: this._factory.fileFormat }
+        : {})
```

Comment:
In the end, I choose to handle it here (rather than [here](https://github.com/jupyterlab/jupyterlab/issues/10757#issuecomment-915784160))  to keep the `null` fileFormat logic at the same place.